### ISSUE TITLE
v2 Breaking Change

### ISSUE THREAD

**Issue by JoleneSF** [google-github-actions/auth#360]
### TL;DR

This action fails now after working a few days ago. The error tells to upgrade to `v2`, and no documentation indicates changes required. However, the upgrade breaks the action workflow.

### Expected behavior

Two possible options are expected for me:
1. The auth works as before.
2. The documentation, changelog, or release notes indicate what needs to be done to fix the breaking change.

### Observed behavior

The original action YAML:

```yml
name: Build and Plan Development

on:
  push:
    branches: [ trunk ]

jobs:
  build_push_dev:
    name: Build Dev
    needs: [ identify ]
    runs-on: ubuntu-latest
    environment: dev
    permissions:
      contents: 'read'
      id-token: 'write'
    steps:
    - uses: actions/checkout@v3
    - uses: actions/setup-node@v3
    - run: npm ci
    - name: Authenticate with Google Cloud
      id: auth_dev
      uses: google-github-actions/auth@v0
      with:
        token_format: access_token
        workload_identity_provider: projects/12345/locations/global/workloadIdentityPools/github-actions/providers/github-actions
        service_account: ABCDEFG@foo.iam.gserviceaccount.com
        access_token_lifetime: 300s
```

Results in the following output:

```sh
# Run google-github-actions/auth@v0
  with:
    token_format: access_token
    workload_identity_provider: projects/12345/locations/global/workloadIdentityPools/github-actions/providers/github-actions
    service_account: ABCDEFG@foo.iam.gserviceaccount.com
    access_token_lifetime: 300s
    create_credentials_file: true
    export_environment_variables: true
    cleanup_credentials: true
    access_token_scopes: https://www.googleapis.com/auth/cloud-platform
    retries: 0
    id_token_include_email: false
Error: The v0 series of google-github-actions/auth is no longer maintained. It will not receive updates, improvements, or security patches. Please upgrade to the latest supported versions: 

    https://github.com/google-github-actions/auth
Created credentials file at "/home/runner/work/tideswell/tideswell/gha-creds-b2edbfb028811c8f.json"
Error: google-github-actions/auth failed with: retry function failed after 1 attempt: failed to generate Google Cloud access token for ABCDEFG@foo.iam.gserviceaccount.com: (403) {
  "error": {
    "code": 403,
    "message": "Permission 'iam.serviceAccounts.getAccessToken' denied on resource (or it may not exist).",
    "status": "PERMISSION_DENIED",
    "details": [
      {
        "@type": "type.googleapis.com/google.rpc.ErrorInfo",
        "reason": "IAM_PERMISSION_DENIED",
        "domain": "iam.googleapis.com",
        "metadata": {
          "permission": "iam.serviceAccounts.getAccessToken"
        }
      }
    ]
  }
}
```

The upgraded action with `google-github-actions/auth@v2` is shown below, along with the new (current) log output.

### Action YAML

```yaml
name: Build and Plan Development

on:
  push:
    branches: [ trunk ]

jobs:
  build_push_dev:
    name: Build Dev
    needs: [ identify ]
    runs-on: ubuntu-latest
    environment: dev
    permissions:
      contents: 'read'
      id-token: 'write'
    steps:
    - uses: actions/checkout@v3
    - uses: actions/setup-node@v3
    - run: npm ci
    - name: Authenticate with Google Cloud
      id: auth_dev
      uses: google-github-actions/auth@v2
      with:
        token_format: access_token
        workload_identity_provider: projects/12345/locations/global/workloadIdentityPools/github-actions/providers/github-actions
        service_account: ABCDEFG@foo.iam.gserviceaccount.com
        access_token_lifetime: 300s
```

### Log output

```text
Error: google-github-actions/auth failed with: retry function failed after 4 attempts: failed to generate Google Cloud OAuth 2.0 Access Token for ABCDEFG@foo.iam.gserviceaccount.com: {
  "error": {
    "code": 403,
    "message": "Permission 'iam.serviceAccounts.getAccessToken' denied on resource (or it may not exist).",
    "status": "PERMISSION_DENIED",
    "details": [
      {
        "@type": "type.googleapis.com/google.rpc.ErrorInfo",
        "reason": "IAM_PERMISSION_DENIED",
        "domain": "iam.googleapis.com",
        "metadata": {
          "permission": "iam.serviceAccounts.getAccessToken"
        }
      }
    ]
  }
}
```

**Comment by nheimann1** [comment]
@JoleneSF 
By looking at the docs it seems you should change the `actions/checkout@v**3**` to `actions/checkout@v**4**`

Have you tried that?

![image](https://github.com/google-github-actions/auth/assets/3142081/6d024ffe-5257-4c2e-ad11-12535128ee52)

**Comment by sethvargo** [comment]
Hi @nheimann1 

Thank you for opening an issue. The only change between 0.8.3 and 0.9.0 is the addition of the message that v0 is deprecated: https://github.com/google-github-actions/auth/compare/v0.8.3...v0.9.0

The error message from the original action.yml indicates there's a misconfiguration with your Workload Identity. Please see the [Troubleshooting guide](https://github.com/google-github-actions/auth/blob/main/docs/TROUBLESHOOTING.md). If none of those steps address your issue, please include your complete action.yml and [debug logs](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/enabling-debug-logging). Please do not sanitize or filter the logs.

I see the same error message in `@v2`, so I don't think this is related to changes in google-github-actions/auth. It appears a configuration may have changed that is causing the authentication to fail.

**Comment by thanhnguyen612** [comment]
Hi @sethvargo ,

I have sent debug logs via email google-github-actions@google.com
Please help to check. Tks.

**Comment by curtis-baillie** [comment]
Thanks @JoleneSF for opening this issue! 
We had the same deprecation error: `Error: The v0 series of google-github-actions/auth is no longer maintained.` 
We upgraded from `v0` to `v1` and that fixed it for us 😃

**Comment by curtis-baillie** [comment]
> @curtis-baillie - it's not an error, it's a log message at the error level. If your GitHub Action is exiting successfully, then you'll see log message, but the GitHub Action will still succeed.

Oh, interesting - I'll have to dig in further to the GHA. It was failing the workflow, and as soon as we changed to `@v1` it succeeded again. Thanks for the follow up!

**Comment by thanhnguyen612** [comment]
Hi @sethvargo ,
I realized that my mistake in changing some terraform code caused our service account to lack permission to get access token from the workload identity pool.

Thanks for your help and sorry.
